internal/credentials/kubernetes/github/app.go

```diff
@@ -198,15 +259,16 @@ func tokenCacheKey(
 	clientID string,
 	installationID int64,
 	encodedPrivateKey string,
+	repoName string,
 ) string {
 	return fmt.Sprintf(
 		"%x",
 		sha256.Sum256([]byte(
 			fmt.Sprintf(
-				"%s:%s:%d:%s",
-				baseURL, clientID, installationID, encodedPrivateKey,
-			),
-		)),
+				"%s:%s:%d:%s:%s",
+				baseURL, clientID, installationID, encodedPrivateKey, repoName),
```

Comment:
The value of `baseURL` is a subset of `repoName`. You don't need it to keep this key unique now that you're including `repoName` in the key.